test/history.test.js

```diff
@@ -20,210 +20,334 @@ describe('session history', () => {
+        /**
+         * Now we're going to make sure that the rule above only applies to large
+         * negative deltas, and not large positive deltas. We don't care about
+         * hanging on to the preceding revision of a large revision, since the
+         * large revision will have more information.
+         */
+        /** Make a small positive change. This should be kept, because it's the last. */
+        item = await setTextAndSync(
+          this.application,
+          item,
+          item.content.text + Factory.randomString(1)
+        );
+        expect(itemHistory.entries.length).to.equal(4);
+        /**
+         * Make a large positive change. The previous small positive change should
+         * now be discarded, keeping a total of 4.
+         */
+        item = await setTextAndSync(
+          this.application,
+          item,
+          item.content.text + Factory.randomString(largeCharacterChange + 1
         ));
-      expect(itemHistory.entries.length).to.equal(4);
+        expect(itemHistory.entries.length).to.equal(4);
+      });
+  });
+  
+  describe('remote', async function () {
+    beforeEach(async function () {
+      this.application = await Factory.createInitAppWithRandNamespace();
+      this.historyManager = this.application.historyManager;
+      this.modelManager = this.application.modelManager;
+      this.email = Uuid.GenerateUuidSynchronously();
+      this.password = Uuid.GenerateUuidSynchronously();
+      await Factory.registerUserToApplication({
+        application: this.application,
+        email: this.email,
+        password: this.password
+      });
+    });
+  
+    afterEach(async function () {
+      await this.application.deinit();
     });
+
+    it('response from server should be empty if not signed in', async function () {
+      await this.application.signOut();
+      this.application = await Factory.createInitAppWithRandNamespace();
+      this.historyManager = this.application.historyManager;
+      this.modelManager = this.application.modelManager;
+      const item = await Factory.createSyncedNote(this.application);
+      await this.application.syncService.sync(syncOptions);
+      const itemHistory = await this.historyManager.serverHistoryForItem(item);
+      expect(itemHistory).to.be.ok;
+      expect(itemHistory.entries.length).to.equal(0);
+    });
+
+    it('create basic history entries', async function () {
+      const item = await Factory.createSyncedNote(this.application);
+      let itemHistory = await this.historyManager.serverHistoryForItem(item);
+
+      /** Server history should not save initial revision */
+      expect(itemHistory).to.be.ok;
+      expect(itemHistory.entries.length).to.equal(0);
+
+      /** Sync with same contents, should not create new entry */
+      await this.application.saveItem(item.uuid);
+      itemHistory = await this.historyManager.serverHistoryForItem(item);
+      expect(itemHistory.entries.length).to.equal(0);
```

Comment:
This assertion is failing. The server returns a revision if the item is saved (with no changes) and synced. Ideally we should save space by not saving a revision if there are no changes.

Reply:
Session history has access to decrypted objects, so can do equality comparison to prevent redundant revisions. However server doesn't have access to decrypted content, so can't do equality comparison. So this is expected, and this should be updated to reflect a length of 1.